File: nodejs/examples/morphology_parts-of-speech.js
/*
 * Example code to call Rosette API to get part-of-speech tags for words a piece of text.
 */

"use strict";

var Api = require("rosette-api").Api;
var ArgumentParser = require("argparse").ArgumentParser;
var DocumentParameters = require("rosette-api").DocumentParameters;
var rosetteConstants = require("rosette-api").rosetteConstants;

var parser = new ArgumentParser({
  addHelp: true,
  description: "Get part-of-speech tags for words in text"
});
parser.addArgument(["--key"], {help: "Rosette API key", required: true});
var args = parser.parseArgs();

var docParams = new DocumentParameters();
var content = "${morphology_parts-of-speech_data}";
docParams.setItem("content", content);

var api = new Api(args.key);
api.morphology(docParams, rosetteConstants.morpholoyOutput.PARTS_OF_SPEECH, function(err, res) {
  if (err) {
    console.log("ERROR! " + err);
  }
  else {
    console.log(res);
  }
});

File: python/examples/morphology_parts-of-speech.py
# -*- coding: utf-8 -*-

"""
Example code to call Rosette API to get part-of-speech tags for words in a piece of text.
"""

import json

from rosette.api import API, DocumentParameters, MorphologyOutput

def run(key):
    # Create an API instance
    api = API(user_key = key)

    params = DocumentParameters()
    params["content"] = u"${morphology_parts_of_speech_data}"
    result = api.morphology(params, MorphologyOutput.PARTS_OF_SPEECH)

    print(json.dumps(result, indent=2, ensure_ascii=False).encode("utf8"))
    return json.dumps(result, indent=2, ensure_ascii=False).encode("utf8")
